File: chapter_38/challenge.c
// Challenge for Chapter 38

#define MAGIC_NUMBER 0x5555555555555555

#include <stdio.h>

int main(int argc, const char * argv[])
{
	unsigned long number = 0;
	number |= MAGIC_NUMBER;
	
	printf("%lu\n", number);
	
	return 0;
}

File: chapter_39/challenge.m
// Challenge for Chapter 39

#include <stdio.h>

int spaceCount(const char* sentence)
{
	int count = 0;
	
	while (*sentence) {
		if (*sentence == 0x20)
			++count;
		++sentence;
	}
	
	return count;
}

int main(int argc, const char * argv[])
{
	const char* sentence = "He was not in the cab at the time.";
	
	printf("\"%s\" has %d spaces.\n", sentence, spaceCount(sentence));
	
	return 0;
}
